### ISSUE TITLE
Incorporate transcription of hyphened <ý> to simple <y> in lemmas on the analysis side

### ISSUE THREAD

**Issue by aarppe** [giellalt/lang-crk#30]
In order to more easily support using the same LEXC source for both crk and cwd (and other Western Cree dialects), we would want to incorporate a simple filter/transcriptor converting the  hyphened `<ý>` to simple `<y>` to the analysis/underlying side of all the FSTs. For Woods Cree, we would similarly convert the hyphened `<ý>` to simple `<th>`. While the coding of `<ý>` is already implemented in the LEXC source on the stem side (for crk), this yet needs to be added to the lemma side of the LEXC code as well.

@snomos The REGEX source for the transcriptor has already been written and added to the repo as: https://github.com/giellalt/lang-crk/blob/main/src/filters/convert-hyphened-y-to-simple-y-in-crk.regex - so what needs to be done is to incorporate this to the appropriate place(s) in the various Makefiles.

**Comment by snomos** [comment]
Can you give some example words to use with testing? What they look like in LexC, what the lemma should be, and what an expected word form should be.

**Comment by aarppe** [comment]
Comprehensive implementations of the LEXC files can be found in `lang-crk/src/fst/incoming`, e.g. https://github.com/giellalt/lang-crk/blob/main/src/fst/incoming/stems/noun_stems.lexc

What I'm suggesting is that to the outside world, nothing changes, so all the YAML tests should pass - it is only the innards that would be different.

**Comment by snomos** [comment]
Sure. I still would need to have an example to test with 😄

**Comment by aarppe** [comment]
All the final FSTs, e.g. the final normative FST analyzer (and the descriptive ones), should work as follows:

`akopayihow -> akopayihow+V+AI+Ind+3Sg`

An interim step, before simplifying `<ý>`, would work as follows (with `<ý>` appearing in the lemma):

`akopayihow -> akopaýihow+V+AI+Ind+3Sg`

**Comment by snomos** [comment]
To recap: the idea is to have a common database for several Cree languages/dialects with abstract/historical forms as the common ground, and generate lexc for each language.

The question is really: what is the best place to do the conversion from `<ý>` to `<y>`? I would argue that it would be best to do the conversion already in the "export" to each language, ie in the generation of the language-specific lexc code. That would give:

- less fst maniputation (and with no actual benefit)
- ensure that lemma generation tests are still working (the tests extract lemmas from lexc, and would thus get `<ý>` in the lemma)

WDYT?

**Comment by aarppe** [comment]
What you're saying is what I originally thought, and that is what the LEXC exportation script currently does. However, not all contemporary `<y>` characters (in Plains), or `<th>` in Woods, are accented ones, so once the conversion is done, it cannot be done deterministically in reverse. In this respect, there could be a benefit in having some FSTs that keep the accented `<ý>`, as well as keeping that on the dictionary content side, and then having as an option whether one wants to see the accented `<ý>` or simple `<y>` or simple `<th>`. Otherwise, in order to be able to present the accented `<ý>` e.g. in itwêwina, we'd need to keep another field with the accented <ý> linked to the lemmas with the regular `<ý>`.

It can be done either way.

**Comment by snomos** [comment]
So the display of the `<ý>` depends on the application/fst? It can not be assumed that all lemmas should always be `<y>`?

**Comment by aarppe** [comment]
Yes. Regular FSTs would have simple `<y>` and `<th>`, but dictionary FSTs would have `<ý>`.
